sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -1703,6 +1704,28 @@ public void populateDisplayData(DisplayData.Builder builder) {
         .build();
   }
 
+  /**
+   * A {@link Sink} for use with {@link FileIO#write} and {@link FileIO#writeDynamic}, writing
+   * elements with a given (common) schema, like {@link #writeGenericRecords(Schema)}.
+   */
+  @Experimental
+  public static <ElementT extends IndexedRecord> Sink<ElementT> sinkWithSchema(Schema schema) {
```

Comment:
What is a point to name this method and `sinkWithSchema(String schema)` below differently and not just overload already existing `sink()`? Not a big deal but imho it could be a bit confusing for users since, anyway, the schema is used by current sink.

Reply:
Good question!  Let's do it.